src/NativeShell/src/RemotePlayerService.java

```diff
@@ -42,13 +45,33 @@
     public static BroadcastReceiver receiver = new BroadcastReceiver() {
         @Override
         public void onReceive(Context context, Intent intent) {
-            if (intent != null && intent.getAction().equals(Intent.ACTION_HEADSET_PLUG)) {
-                int state = intent.getIntExtra("state", 2);
-                if (state == 0) {
-                    sendCommand("playpause");
-                    headphoneFlag = true;
-                } else if (headphoneFlag) {
-                    sendCommand("playpause");
+            if (intent != null) {
```

Comment:
I would just add `if (intent == null) continue` on the line above to reduce nesting.